core/src/main/scala/org/apache/spark/Accumulator.scala

```diff
@@ -75,43 +84,65 @@ private[spark] object Accumulators extends Logging {
    * This global map holds the original accumulator objects that are created on the driver.
    * It keeps weak references to these objects so that accumulators can be garbage-collected
    * once the RDDs and user-code that reference them are cleaned up.
+   * TODO: Don't use a global map; these should be tied to a SparkContext at the very least.
    */
+  @GuardedBy("Accumulators")
```

Comment:
+1 on this style of explicit documentation of locking / synchronization requirements. This is a good practice and we should do this more often.